suite2p/io/utils.py

```diff
@@ -90,103 +92,113 @@ def list_files(froot, look_one_level_down, exts):
+        print("** Found %d tifs - converting to binary **" % (len(fsall)))
     else:
-        if len(froot)==1:
-            if 'subfolders' in ops and len(ops['subfolders'])>0:
+        if len(froot) == 1:
+            if "subfolders" in ops and len(ops["subfolders"]) > 0:
                 fold_list = []
-                for folder_down in ops['subfolders']:
+                for folder_down in ops["subfolders"]:
                     fold = os.path.join(froot[0], folder_down)
                     fold_list.append(fold)
             else:
-                fold_list = ops['data_path']
+                fold_list = ops["data_path"]
         else:
             fold_list = froot
         fsall = []
         nfs = 0
         first_tiffs = []
-        for k,fld in enumerate(fold_list):
-            fs, ftiffs = list_files(fld, ops['look_one_level_down'],
-                                    ["*.tif", "*.tiff", "*.TIF", "*.TIFF"])
+        for k, fld in enumerate(fold_list):
+            fs, ftiffs = list_files(
+                fld, ops["look_one_level_down"], ["*.tif", "*.tiff", "*.TIF", "*.TIFF"]
+            )
             fsall.extend(fs)
             first_tiffs.extend(list(ftiffs))
-        if len(fsall)==0:
-            print('Could not find any tiffs')
-            raise Exception('no tiffs')
+        if len(fsall) == 0:
+            print("Could not find any tiffs")
+            raise Exception("no tiffs")
         else:
-            ops['first_tiffs'] = np.array(first_tiffs).astype('bool')
-            print('** Found %d tifs - converting to binary **'%(len(fsall)))
+            ops["first_tiffs"] = np.array(first_tiffs).astype("bool")
+            print("** Found %d tifs - converting to binary **" % (len(fsall)))
     return fsall, ops
 
 
 def get_nd2_list(ops):
-    """ make list of nd2 files to process
-    if ops['look_one_level_down'], then all nd2's in all folders + one level down
     """
-    froot = ops['data_path']
-    fold_list = ops['data_path']
-    fsall = []
-    nfs = 0
-    first_tiffs = []
-    for k,fld in enumerate(fold_list):
-        fs, ftiffs = list_files(fld, ops['look_one_level_down'],
-                                ["*.nd2"])
-        fsall.extend(fs)
-        first_tiffs.extend(list(ftiffs))
-    if len(fs)==0:
-        print('Could not find any nd2 files')
-        raise Exception('no nd2s')
+    if tiff_list is specified in ops, construct list of .nd2 files specified
+    else construct a list of all nd2 files to process
+        - if ops['look_one_level_down'], then obtain all nd2's in all folders + one level down
+    """
+    froot = ops["data_path"]
+    if "tiff_list" in ops:
+        fsall = []
+        for tif in ops["tiff_list"]:
+            fsall.append(os.path.join(froot[0], tif))
+        ops["first_tiffs"] = np.zeros((len(fsall),), dtype="bool")
+        ops["first_tiffs"][0] = True
+        print("** Found %d nd2 files - converting to binary **" % (len(fsall)))
```

Comment:
This is duplicated again below, you can just move this out of the if/else block.